src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ProvideCorrectArgumentsToFormattingMethodsTests.cs

```diff
@@ -510,6 +510,101 @@ End Sub
             await basicTest.RunAsync();
         }
 
+        [Fact]
+        [WorkItem(6012, "https://github.com/dotnet/roslyn-analyzers/issues/6012")]
+        public async Task EditorConfigConfiguration_StringSyntaxAnnotatedMethodsAsync()
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+using System.Diagnostics.CodeAnalysis;
+
+class Test
+{
+    public static string MyFormat([StringSyntax(StringSyntaxAttribute.CompositeFormat)] string specification, params object[] args) => specification;
+
+    void M1(string param)
+    {
+        var a = MyFormat("""", 1);
+    }
+}",
+                        @"
+namespace System.Diagnostics.CodeAnalysis
+{
+    /// <summary>Specifies the syntax used in a string.</summary>
+    [AttributeUsage(AttributeTargets.Parameter | AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = false)]
+    public sealed class StringSyntaxAttribute : Attribute
```

Comment:
You could reference .NET 7 instead of adding the attribute, see:
https://github.com/dotnet/roslyn-analyzers/blob/114ed7b80e68f6d830fff399f7b2c8f3367d58d5/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedTests.cs#L1093

Reply:
Thanks. Done.